src/hooks/useActiveProject.tsx

```diff
@@ -49,6 +51,46 @@ const findProjectAndSubjectById = (
   return { selectedProject, selectedSubject };
 };
 
+/**
+ * Why an HOC:
+ *
+ * Using a higher-order component here allows us to more easily (and type-safely)
+ * model the GET /v1/accounts network dependencies here. The idea is:
+ *
+ * - Until we're sure the user is in the active account, it is not safe to make other
+ *   network requests.
+ *   -> (because if they aren't in the account, the request will result in 401)
+ *
+ * - So, rather than deal with passing a react-query `enabled` setting through the various
+ *   project-related network calls, just avoid rendering those hooks entirely until we
+ *   know the user is in the right account.
+ */
+
+const withAccountRequired =
+  <Props extends {}>(Component: React.FC<Props>): React.FC<Props> =>
```

Comment:
@aecorredor I just slimmed down the # of HOCs. This one is in place because it makes modeling the `accounts -> projects` dependency much clearer. Without this HOC, we'd need to set a `react-query` `enabled` state on each of the calls to `useUser`, `useSubjectProjects`, and `useMe` in the provider below. This made the modeling much clearer.